### ISSUE TITLE
SC2094: Using sort -n  and uniq

### ISSUE THREAD

**Issue by sahil-lakhwani** [koalaman/shellcheck#1288]
Rule Id: SC2094
`shellcheck -v : 0.5.0`

```
#!/bin/bash

/bin/cut -f3 -d":" < /etc/passwd | /bin/sort -n | /usr/bin/uniq -c |\
    while read x ; do
    [ -z "${x}" ] && break
    # shellcheck disable=SC2086
    set - $x    
    if [ "$1" -gt 1 ]; then
        users=$(/bin/gawk -F: '($3 == n) { print $1 }' n="$2" \
            /etc/passwd | /usr/bin/xargs)
        echo "Duplicate UID ($2): ${users}"
fi done
```

shellcheck complains:

```
/bin/cut -f3 -d":" < /etc/passwd | /bin/sort -n | /usr/bin/uniq -c |\
                     ^-- SC2094: Make sure not to read and write the same file in the same pipeline.
```
and

```
/etc/passwd | /usr/bin/xargs)
            ^-- SC2094: Make sure not to read and write the same file in the same pipeline.
```

Do `sort` & `uniq` act on the file again or they act on the result of `cut` ?

**Comment by matthewpersico** [comment]
They act on the result of cut. I'm with you on this one. But note that there is no error printed if you paste this into the checker at shellcheck.net

**Comment by sahil-lakhwani** [comment]
@matthewpersico Yes it doesn't give errors when checked on shellchecker.net

I think the issue is something else. Apologies for not posting the complete code. I have edited the sample code in the issue now.

**Comment by matthewpersico** [comment]
Ah! Now it makes sense. You are reading /etc/passwd to feed items to a while loop in which you again read /etc/passwd. Now you and I can both see that the file is only being read in this case, but I am betting that the analysis needed to determine that in shellcheck is not available. Hence, the precautionary warning. I suggest you drop in a 2094 exclusion and move on for now. Let's see what @koalaman has to say.

**Comment by koalaman** [comment]
You're entirely right, and your code is correct. ShellCheck is just really bad at determining whether a mentioned file is read, written or neither. Please just exclude it. Sorry it took a while. 

It's generally a tricky problem: compare the role of `a_file` in `cat a_file`, `tee a_file` and `echo a_file`. Then try `sed '' a_file`,  `sed -i '' a_file` and `sed a_file`. 

SC2094 just triggers whenever the target of a redirection also appears in a command in the pipeline. This aims to catch things like `cat file > file` , but results in a false positive for your command, and a false negative for things like `cat foo | tee foo`.
